doc/OnlineDocs/contributed_packages/incidence/tutorial.bt.rst

```diff
@@ -0,0 +1,101 @@
+Debugging a numeric singularity using block triangularization
+=============================================================
+
+We start with some imports. To debug a *numeric* singularity, we will need
+``PyomoNLP`` from :ref:`pynumero` to get the constraint Jacobian,
+and will need NumPy to compute condition numbers.
+
+.. doctest::
+   :skipif: not scipy_available or not asl_available or not networkx_available
+
+   >>> import pyomo.environ as pyo 
+   >>> from pyomo.contrib.pynumero.interfaces.pyomo_nlp import PyomoNLP
+   >>> from pyomo.contrib.incidence_analysis import IncidenceGraphInterface
+   >>> import numpy as np
+
+We now build the model we would like to debug. Compared to the model in
+:ref:`incidence_tutorial_dm`, we have converted the sum equation to use a sum
+over component flow rates rather than a sum over mass fractions.
+
+.. doctest::
+   :skipif: not scipy_available or not asl_available or not networkx_available
+
+   >>> m = pyo.ConcreteModel()
+   >>> m.components = pyo.Set(initialize=[1, 2, 3]) 
+   >>> m.x = pyo.Var(m.components, initialize=1.0/3.0)
+   >>> m.flow_comp = pyo.Var(m.components, initialize=10.0)
+   >>> m.flow = pyo.Var(initialize=30.0)
+   >>> m.density = pyo.Var(initialize=1.0)
+   >>> # This equation is new!
+   >>> m.sum_flow_eqn = pyo.Constraint(
+   ...     expr=sum(m.flow_comp[j] for j in m.components) == m.flow
+   ... )
+   >>> m.holdup_eqn = pyo.Constraint(m.components, expr={
+   ...     j: m.x[j]*m.density - 1 == 0 for j in m.components
+   ... })
+   >>> m.density_eqn = pyo.Constraint(
+   ...     expr=1/m.density - sum(1/m.x[j] for j in m.components) == 0
+   ... )
+   >>> m.flow_eqn = pyo.Constraint(m.components, expr={
+   ...     j: m.x[j]*m.flow - m.flow_comp[j] == 0 for j in m.components
+   ... })
+
+We now construct the incidence graph and check unmatched variables and
+constraints to validate structural nonsingularity.
+
+.. doctest::
+   :skipif: not scipy_available or not asl_available or not networkx_available
+
+   >>> igraph = IncidenceGraphInterface(m, include_inequality=False)
+   >>> var_dmp, con_dmp = igraph.dulmage_mendelsohn()
+   >>> print(len(var_dmp.unmatched))
+   0
+   >>> print(len(con_dmp.unmatched))
+   0
+
+Our system is structurally nonsingular. Now we check whether we are numerically
+singular (well-conditioned) by checking the condition number.
```

Comment:
This implies that numerically singular systems are well-conditioned.  I think you want to be checking that `we are numerically nonsingular`?